### ISSUE TITLE
pS Testpoint install: chrony is being uninstalled in favor of ntp

### ISSUE THREAD

**Issue by tonin** [perfsonar/bundles#15]
@igarny commented on [Thu Apr 12 2018](https://github.com/perfsonar/toolkit/issues/308)

along with the inquiry here: #278
GEANT Operations have switched to using Chrony for the time synchronization on the servers.
It is even currently running on the production CentOS 6 servers and in order to get around the problem we are disabling the ntp daemon. This configuration is working already for about 2 months.

Would it be possible to possible to support Chrony  along with ntp daemon?

Here is the output on an Ubuntu16 perfsonar-testpoint install 
`perfsonar-testpoint (4.0.2 psmp/test perfsonar-jessie-release:perfsonar-jessie-release [all]) `

```
The following packages were automatically installed and are no longer required:
The following additional packages will be installed:
  libnet-ipv6addr-perl libnet-netmask-perl libnet-ntp-perl libnet-server-perl
  libyaml-syck-perl memcached netpbm ntp nuttcp owamp-client owamp-server
  python-memcache python-mock python-netaddr python-netifaces python-ntplib
  libyaml-perl ntp-doc apparmor poppler-utils fonts-japanese-mincho
The following packages will be REMOVED:
  chrony
The following NEW packages will be installed:
  libnet-ipv6addr-perl libnet-netmask-perl libnet-ntp-perl libnet-server-perl
  libyaml-syck-perl memcached netpbm ntp nuttcp owamp-client owamp-server
  python-memcache python-mock python-netaddr python-netifaces python-ntplib
The following packages will be upgraded:
```

---

@arlake228 commented on [Thu Apr 12 2018](https://github.com/perfsonar/toolkit/issues/308#issuecomment-380798103)

The reason we default to NTP is for the toolkit GUI since it needs to be updated to update chrony config files instead of NTP and right now, and that's not considered high priority at the moment. Since you are not using the toolkit GUI to configure NTP, you should be fine using chrony.

Closing since this also duplicates #278

---

@igarny commented on [Thu Apr 12 2018](https://github.com/perfsonar/toolkit/issues/308#issuecomment-380805035)

Re opened since pS Testpoint install uninstalls `chrony` in favor of `ntp` package.
possibly both can be supported on non-toolkit deployments

---

@arlake228 commented on [Thu Apr 12 2018](https://github.com/perfsonar/toolkit/issues/308#issuecomment-380805478)

Also noting this has only been observed on Ubuntu16, other OSes would need to be checked.

---

@tonin commented on [Fri Apr 13 2018](https://github.com/perfsonar/toolkit/issues/308#issuecomment-381108801)

In the Debian/Ubuntu packages, from the **perfsonar-testpoint** bundle package, the ntp dependency is registered in a single place only: **bwctl**.  Here is the full reference list:
 - bwctl has a direct dependency to `ntp`
 - oppd has a direct dependency to `ntp`, but oppd is not installed by default anymore
 - owamp has a conditional dependency: `ntp | time-daemon`
 - libperfsonar-perl has a dependency to `libnet-ntp-perl` but that should not bring a dependency on ntp
 - python-pscheduler has a dependency to `python-ntplib` but that should not bring a dependency on ntp
 - perfsonar-toolkit and perfsonar-toolkit-ntp have a direct dependency to `ntp` but they shouldn't be required by the testpoint bundle.

The only place where the **perfsonar-testpoint** bundle still has a hard dependency on  ntp is through **bwctl**.  This should go away with 4.1.

Meanwhile, is there anything we can do?  I'm not sure about the internals of **bwctl**, but does it rely on ntp itself or just on good time-sync?  Could we change the dependency to `ntp | time-daemon` like it is listed for owamp?

**Comment by tonin** [comment]
I think it would be safe to remove the ntp hard dependency from the **bwctl** Debian packages and that should solve this **perfsonar-testpoint** bundle installation issue.  I've created an [issue to track that](https://github.com/perfsonar/bwctl/issues/44).

**Comment by tonin** [comment]
@igarny I've published new bwctl packages to the staging repo.  Can you try using http://downloads.perfsonar.net/debian/perfsonar-jessie-staging.list as an additional apt source file and see if you can keep chrony with that?

**Comment by igarny** [comment]
@tonin 
on staging I still observe the problem:
```
ps2:/var/log# ls /etc/apt/sources.list.d/perfsonar-jessie-*
/etc/apt/sources.list.d/perfsonar-jessie-release.list
/etc/apt/sources.list.d/perfsonar-jessie-staging-release.list
```

```
ps2:/var/log# apt-get -s install chrony
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following packages were automatically installed and are no longer required:
  aglfn apache2 apache2-bin apache2-data apache2-utils bwctl-client
  bwctl-server bzip2 cpp cpp-5 dbconfig-common docutils-common docutils-doc
  ethtool fontconfig fontconfig-config fonts-dejavu-core fonts-liberation
  geoip-database-extra ghostscript gnuplot-tex gnuplot5-data gnuplot5-nox
  groff groff-base gsfonts hicolor-icon-theme ieee-data imagemagick
  imagemagick-6.q16 imagemagick-common iperf iperf3 iputils-ping
  iputils-tracepath iso-codes javascript-common jq libalgorithm-c3-perl
  libapache2-mod-wsgi libappconfig-perl libapr1 libaprutil1
  libaprutil1-dbd-sqlite3 libaprutil1-ldap libapt-inst2.0 libauthen-sasl-perl
  libavahi-client3 libavahi-common-data libavahi-common3
  libb-hooks-endofscope-perl libcairo2 libcgi-ajax-perl libcgi-fast-perl
.................................................................................
  python-memcache python-mock python-ndg-httpsclient python-netaddr
  python-netifaces python-ntplib python-openssl python-pam python-pbr
  python-pika python-pil python-pscheduler python-psycopg2 python-pyasn1
  python-pyasn1-modules python-pygments python-pyinotify python-pyjq
  python-pysnmp4 python-pysnmp4-apps python-pysnmp4-mibs python-radix
  python-requests python-roman python-serial python-service-identity
  python-six python-subprocess32 python-twisted python-twisted-bin
  python-twisted-core python-tz python-tzlocal python-urllib3 python-werkzeug
  python-zope.interface rename shared-mime-info smitools sudo x11-common
  xdg-user-dirs
Use 'apt autoremove' to remove them.
The following additional packages will be installed:
  install-info libtomcrypt0 libtommath0 timelimit
The following packages will be REMOVED:
  ntp perfsonar-oppd-bwctl perfsonar-oppd-owamp perfsonar-oppd-server
  perfsonar-testpoint
The following NEW packages will be installed:
  chrony install-info libtomcrypt0 libtommath0 timelimit
0 upgraded, 5 newly installed, 5 to remove and 0 not upgraded.
Remv perfsonar-testpoint [4.0.2]
Remv perfsonar-oppd-bwctl [4.0-1]
Remv perfsonar-oppd-owamp [4.0-1]
Remv perfsonar-oppd-server [4.0-1]
Remv ntp [1:4.2.8p4+dfsg-3ubuntu5.8] [owamp-server:amd64 ]
Inst chrony (2.1.1-1 Ubuntu:16.04/xenial [amd64]) []
Inst install-info (6.1.0.dfsg.1-5 Ubuntu:16.04/xenial [amd64]) []
Conf install-info (6.1.0.dfsg.1-5 Ubuntu:16.04/xenial [amd64]) []
Inst timelimit (1.8-1 Ubuntu:16.04/xenial [amd64]) []
Inst libtommath0 (0.42.0-1.2 Ubuntu:16.04/xenial [amd64]) []
Inst libtomcrypt0 (1.17-7 Ubuntu:16.04/xenial [amd64])
Conf libtommath0 (0.42.0-1.2 Ubuntu:16.04/xenial [amd64])
Conf libtomcrypt0 (1.17-7 Ubuntu:16.04/xenial [amd64])
Conf timelimit (1.8-1 Ubuntu:16.04/xenial [amd64])
Conf chrony (2.1.1-1 Ubuntu:16.04/xenial [amd64])

```

**Comment by tonin** [comment]
Ok, thanks for this test.  I now see I was wrong stating OPPD was not installed by default, it is still being installed with a 4.0.2 pS, just not enabled as a service.  And so this brings back ntpd.

**Comment by tonin** [comment]
This has been solved for the upcoming major release: 4.1  We will not do a fix for 4.0.

**Comment by igarny** [comment]
Actually I decided to open a new issue, since this one speaks about the software conflict only
